File: carnerf-mobile/lib/constants/warranty.ts
export interface WarrantyTerms {
  general: { years: number; km: number };
  powertrain: { years: number; km: number };
  rust?: { years: number; km: number };
}

const UNLIMITED_KM = 999_999;

export const WARRANTY_BY_BRAND: Record<string, WarrantyTerms> = {
  현대: {
    general: { years: 3, km: 60000 },
    powertrain: { years: 5, km: 100000 },
    rust: { years: 5, km: UNLIMITED_KM },
  },
  기아: {
    general: { years: 3, km: 60000 },
    powertrain: { years: 5, km: 100000 },
    rust: { years: 5, km: UNLIMITED_KM },
  },
  제네시스: {
    general: { years: 5, km: 100000 },
    powertrain: { years: 5, km: 100000 },
    rust: { years: 7, km: UNLIMITED_KM },
  },
  쉐보레: {
    general: { years: 5, km: 100000 },
    powertrain: { years: 5, km: 100000 },
  },
  BMW: {
    general: { years: 3, km: UNLIMITED_KM },
    powertrain: { years: 3, km: UNLIMITED_KM },
  },
  벤츠: {
    general: { years: 3, km: UNLIMITED_KM },
    powertrain: { years: 3, km: UNLIMITED_KM },
  },
  아우디: {
    general: { years: 3, km: UNLIMITED_KM },
    powertrain: { years: 3, km: UNLIMITED_KM },
  },
  렉서스: {
    general: { years: 4, km: 100000 },
    powertrain: { years: 6, km: 120000 },
  },
  포르쉐: {
    general: { years: 4, km: UNLIMITED_KM },
    powertrain: { years: 4, km: UNLIMITED_KM },
  },
};

export const INSPECTION_WARRANTY = { days: 30, km: 2000 };

File: carnerf-flutter/lib/core/constants/warranty.dart
class WarrantyTerm {
  const WarrantyTerm({required this.years, required this.km});
  final int years;
  final int km;
}

class WarrantyTerms {
  const WarrantyTerms({
    required this.general,
    required this.powertrain,
    this.rust,
  });

  final WarrantyTerm general;
  final WarrantyTerm powertrain;
  final WarrantyTerm? rust;
}

const int kUnlimitedKm = 999999;

const Map<String, WarrantyTerms> warrantyByBrand = {
  '현대': WarrantyTerms(
    general: WarrantyTerm(years: 3, km: 60000),
    powertrain: WarrantyTerm(years: 5, km: 100000),
    rust: WarrantyTerm(years: 5, km: kUnlimitedKm),
  ),
  '기아': WarrantyTerms(
    general: WarrantyTerm(years: 3, km: 60000),
    powertrain: WarrantyTerm(years: 5, km: 100000),
    rust: WarrantyTerm(years: 5, km: kUnlimitedKm),
  ),
  '제네시스': WarrantyTerms(
    general: WarrantyTerm(years: 5, km: 100000),
    powertrain: WarrantyTerm(years: 5, km: 100000),
    rust: WarrantyTerm(years: 7, km: kUnlimitedKm),
  ),
  '쉐보레': WarrantyTerms(
    general: WarrantyTerm(years: 5, km: 100000),
    powertrain: WarrantyTerm(years: 5, km: 100000),
  ),
  'BMW': WarrantyTerms(
    general: WarrantyTerm(years: 3, km: kUnlimitedKm),
    powertrain: WarrantyTerm(years: 3, km: kUnlimitedKm),
  ),
  '벤츠': WarrantyTerms(
    general: WarrantyTerm(years: 3, km: kUnlimitedKm),
    powertrain: WarrantyTerm(years: 3, km: kUnlimitedKm),
  ),
  '아우디': WarrantyTerms(
    general: WarrantyTerm(years: 3, km: kUnlimitedKm),
    powertrain: WarrantyTerm(years: 3, km: kUnlimitedKm),
  ),
  '렉서스': WarrantyTerms(
    general: WarrantyTerm(years: 4, km: 100000),
    powertrain: WarrantyTerm(years: 6, km: 120000),
  ),
  '포르쉐': WarrantyTerms(
    general: WarrantyTerm(years: 4, km: kUnlimitedKm),
    powertrain: WarrantyTerm(years: 4, km: kUnlimitedKm),
  ),
};

class InspectionWarranty {
  const InspectionWarranty({required this.days, required this.km});
  final int days;
  final int km;
}

const InspectionWarranty inspectionWarranty = InspectionWarranty(days: 30, km: 2000);
